### ISSUE TITLE
Hex to string and new lines

### ISSUE THREAD

**Issue by ProgerXP** [ProgerXP/Notepad2e#123]
I have a document:
```
1
2
3
```
using Windows line breaks (but the problem occurs with any).

I do Ctrl+Alt+A, then Ctrl+Alt+Shift+A which should result in the original document. But it doesn't.

**Comment by cshnik** [comment]
HexToString/StringToHex works as stated in #87  

>  * encode whitespace (<=0x20) as is; on decode ignore (i.e. space = `20` but on Hex->Char spaces are ignored).
  
and  
  
>For example:
>```
>abc
>def
>```
>Gives:
>```
>6162630D0A646566
>```

**Comment by ProgerXP** [comment]
>encode whitespace (<=0x20) as is; on decode ignore (i.e. space = 20 but on Hex->Char spaces are ignored).

You have misunderstood the requirement. "on decode" refers to **literal** whitespace, not encoded whitespace, otherwise this would be useless.

When encoding, 0x20, 0x0A, 0x09 and other characters with ASCII code <= 0x20 are encoded as is - same as any character with ASCII code > 0x20. So a space gives "20" in the output.

When decoding, literal (raw) characters like spaces and line breaks (with ASCII code <= 0x20) are ignored. This is because such characters may be added for readability, transmission or other requirements. For example, AFAIR one of the Base64 standards (MIME) specifies that base64-encoded string must be split at 76 column. But when decoding such a string, line breaks are ignored.

This does **not** mean that line breaks encoded in base64 are ignored! It doesn't make sense.

In other words:
1. Source:
```
6162630D0A646566
```
Result:
```
abc
def
```
2. Source:
```
61 62 63
0D 0A
64 65 66
```
Result:
```
abc
def
```
3. Source:
```
61 62 63

64 65 66
```
Result:
```
abcdef
```

**Comment by cshnik** [comment]
Fixed.

**Comment by ProgerXP** [comment]
Memo regarding 6be5ba7: the TestHex_FileSamples test was erratically failing due to using randomized buffer size. After investigation an error was found in StrToHex implementation.
